### ISSUE TITLE
improvement on note management system

### ISSUE THREAD

**Issue by JaakDragon** [astudentinearth/darkwrite#23]
The bi-directional note management system doesn't feels right, it lacks the accessibility and feel like siyuan note on how they handle the bi-directional note management system and how the note management is fluid (not in the sense of speed or animation, this app has better, but how the UX feels).

I would love to see improvement in this sector as its important, cause the users will be jumping from one note to the other and linking notes from all around their workspace.

**Comment by astudentinearth** [comment]
I found SiYuan a bit overwhelming, if I'm totally honest. The UI is confusing, that's part of the reason why I don't rush features here because I don't want Darkwrite to become a confusing tool. It sometimes feels confusing/inaccessible now because it's still kind of minimal.

The good news I can give here is that I drafted the initial plan for databases (#13) and once it comes it'll help a lot. In the first pass only table/list view will be available without manual sorting (if I enter this rabbit hole even tables will get severely delayed). This will be the first time 2 editors will be displayed at once, and once we cross this river I plan to add tabs as soon as possible.

Search (⌘K) is kinda dysfunctional now too, it only checks for a match and doesn't do any algorithmic sorting. I'm sure this is slowing down navigation too, at least it did on my end.

Graph view (#17) is still on the table, I just need to figure out how to build it without choking the app. The latest update was mostly about not loading all notes at once and not creating a mess, but a graph requires a completely separate index for relationships, and we need to traverse document trees to find links.

The sidebar is, unfortunately, really bad when it comes to keyboard navigation. You have to tab your way through. I want to bring in arrow keys but even if it sounds simple - there are still many technical considerations.

All of these in isolation sound simple, but without proper thinking they could easily bottleneck future updates, that's why asking AI to do it isn't an option either. Even with databases I ended up documenting a dozen edge cases and interactions. I'd miss all these cases if I let an AI code it because it generates perfect-looking code for the happy path, but ignores all the nuances.

The more I work on this app, the more I realize why the existing alternatives aren't that great. It's just so difficult :/

**Comment by JaakDragon** [comment]
Love your Hard Work and how you want minimal but fulfill app before it leaves the beta stage
